deploy-queue/src/main.rs

```diff
@@ -0,0 +1,379 @@
+                Ok(id) => id,
+                Err(e) => {
+                    log::error!("Failed to insert deployment record: {}", e);
+                    anyhow::bail!("Database insertion failed: {}", e);
+                }
+            };
+
+            loop {
+                // Check for blocking deployments in the same region
+                let blocking_deployments = check_blocking_deployments(&db_client, deployment_id).await?;
+                
+                if blocking_deployments.is_empty() {
+                    info!("No blocking deployments found. Deployment can be started.");
+
+                    // Update deployment record to set start_timestamp
+                    match start_deployment(&db_client, deployment_id).await {
+                        Ok(()) => {
+                            info!("Successfully started deployment with ID: {}", deployment_id);
+                            break;
+                        }
+                        Err(e) => {
+                            log::error!("Failed to start deployment: {}", e);
+                            anyhow::bail!("Database update failed: {}", e);
+                        }
+                    }
+                } else {
+                    // Print information about blocking deployments
+                    info!("Found {} blocking deployment(s) with smaller queue positions:", 
+                        blocking_deployments.len());
+                    for pending_deployment in &blocking_deployments {
+                        let deployment_state: DeploymentState = pending_deployment.into();
+                        let deployment_note = pending_deployment.url.or(pending_deployment.note).unwrap_or_else(|| String::new());
+                        info!("  - Deployment ID: {}, Component: {}, State: {}, Note: {}", pending_deployment.id, pending_deployment.component, deployment_state, deployment_note);
+                    }
+                    info!("Retrying in 5 seconds.");
+                    sleep(BUSY_RETRY).await;
+                }
+            }
+        }
+        Mode::Finish { deployment_id } => {
+            log::info!("Finishing deployment with ID: {}", deployment_id);
+
+            // Verify deployment exists and is in a valid state for finishing
+            match verify_deployment_can_be_finished(&db_client, *deployment_id).await {
+                Ok(true) => {
+                    // Update deployment record to set finish_timestamp
+                    match finish_deployment(&db_client, *deployment_id).await {
+                        Ok(()) => {
+                            log::info!("Successfully finished deployment with ID: {}", deployment_id);
+                        }
+                        Err(e) => {
+                            log::error!("Failed to update deployment record: {}", e);
+                            anyhow::bail!("Database update failed: {}", e);
+                        }
+                    }
```

Comment:
```suggestion
                    finish_deployment(&db_client, *deployment_id).await
                        .context("Failed to set deployment to finished")?;
                    log::info!("Successfully finished deployment with ID: {}", deployment_id);
```

Same here, this is a lot more verbose than it needed to be.